src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -98,12 +98,63 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
     public void createAddGameFromRoom(Room room) {
         Game game = GameMapper.fromRoom(room);
+        Long time = room.getDuration();
 
         List<Problem> problems = problemService.getProblemsFromDifficulty(room.getDifficulty(), room.getNumProblems());
         game.setProblems(problems);
-        setStartGameTimer(game, room.getDuration());
+        setStartGameTimer(game, time);
 
         currentGameMap.put(room.getRoomId(), game);
+
+        // Create notifications for different "time left" milestones.
+        if (GameTimer.DURATION_60 < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are sixty minutes");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_60) * 1000);
+        }
+        
+        if (GameTimer.DURATION_30 < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are thirty minutes");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_30) * 1000);
+        }
+        
+        if (GameTimer.DURATION_15 < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are fifteen minutes");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_15) * 1000);
+        }
+        
+        if (GameTimer.DURATION_5 < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are five minutes");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_5) * 1000);
+        }
+        
+        if (GameTimer.DURATION_1 < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "is one minute");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_1) * 1000);
+        }
+
+        if (GameTimer.DURATION_30_SEC < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are thirty seconds");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_30_SEC) * 1000);
+        }
+
+        if (GameTimer.DURATION_10_SEC < time) {
+            Timer timer = new Timer();
+            NotificationTimerTask notificationTimerTask =
+                new NotificationTimerTask(socketService, room.getRoomId(), "are ten seconds");
+            timer.schedule(notificationTimerTask, (time - GameTimer.DURATION_10_SEC) * 1000);
+        }
```

Comment:
Is it possible to simplify this with loops? Not sure the exact way it would work, but I could see like having a static list of the `[DURATION_X]` values (which we could then easily customize if we wanted to change notification frequencies). There might be some issues with the "are X min" messages, but I feel maybe we could get around that by changing the message to some other less rigid form like "Time remaining in this game is Xmin" (X might have to be in number format to allow it to work with the loop, e.g. `DURATION / 60` to get the correct minute count). Although this would only work with Xmin and not Xsecs notifications (but I think down to the last minute we could also just have the timer flashing red or something rather than a notification with only like 10 seconds left)

That paragraph might make it seem very complicated tho haha, so I'm open to talking about this to see whether it actually makes sense or not

Reply:
I updated this to use a for loop before this comment! 🎉  You can see my implementation, I used a map between the duration and content -- let me know if you think I should change anything there.